### DISCUSSION TITLE
POMDPs with state-dependent actions

### DISCUSSION THREAD

**Discussion by soheilmn** [JuliaPOMDP/POMDPs.jl#353] [Q&A] [answered]
Hi everyone, I am new to Julia and JuliaPOMDP package. In my POMDP model, certain actions may not available at specific states, and I was wondering if there is any example in the repositories that may be instructive on implementation of state-dependent actions. Thanks for your help!

**Comment by lassepe** [comment, marked as answer]
You can implement the two-argument version `POMDPs.actions(model, s)`, where the second argument is a state (in an MDP) or a belief (in a POMDP). Using this dispatch you can return different action spaces for different states/beliefs. See also this discussion: https://github.com/JuliaPOMDP/POMDPs.jl/discussions/338

**Reply by soheilmn** [reply]
Thank you!
